workspaces/ai-integrations/plugins/catalog-backend-module-model-catalog/src/clients/ModelCatalogGenerator.ts

```diff
@@ -189,7 +204,33 @@ export function GenerateModelServerAPI(
   };
 
   if (api.tags !== undefined) {
-    modelServerAPIEntity.metadata.tags = api.tags;
+    modelServerAPIEntity.metadata.tags = sanitizeTags(api.tags, logger);
   }
   return modelServerAPIEntity;
 }
+
+function sanitizeMetadataName(modelName: string): string {
+  return modelName.replace(/\s/g, '');
+}
+
+function sanitizeTags(tags: string[], logger?: LoggerService): string[] {
+  const sanitizedTags: string[] = [];
+  tags.forEach(tag => {
+    let sanitizedTag: string = tag;
+    // Replace whitespace with empty character
+    sanitizedTag = sanitizedTag.replace(/\s/g, '');
+
+    // Call the Backstage tag validator and check if the tag conforms
+    // If the tag is not valid, skip it
+    if (!makeValidator().isValidTag(sanitizedTag)) {
+      if (logger !== undefined) {
+        logger.error(
+          `invalid tag: ${sanitizedTag}. Tags are expected to be less than 63 characters and conform to: ^[a-z0-9:+#]+(\-[a-z0-9:+#]+)*$`,
```

Comment:
don't have the backstage code up, but is it possible to pull the regexp string from `makeValidator` or whatever constant that actual regex string constant is defined 

if so, let's reference that constant etc. vs. hard coding the regex string in your log message

Reply:
Looks like the Regexp string isn't defined as a constant in makeValidator: https://github.com/backstage/backstage/blob/9ef2f1e08a64544575dedc94d39c902b4083a4b3/packages/catalog-model/src/validation/makeValidator.ts#L35

I'll leave it as is for now.

Reply:
sounds good